### PR TITLE
Adds flush API

### PR DESCRIPTION
**Pull request by peterkrull**
Adds the ability to manually flush/commit written data to the directory entry, as discussed in #98.

### CODE REVIEW

File: src/filesystem/files.rs [hunk 1]
```diff
@@ -138,6 +138,18 @@ where
...
 }
 
+impl<'a, D, T, const MAX_DIRS: usize, const MAX_FILES: usize, const MAX_VOLUMES: usize>
+    File<'a, D, T, MAX_DIRS, MAX_FILES, MAX_VOLUMES>
+where
+    D: crate::BlockDevice,
+    T: crate::TimeSource,
+{
+    /// Flush any written data by updating the directory entry.
+    pub fn flush(&mut self) -> Result<(), Error<D::Error>> {
```
Comment (thejpster) [inline src/filesystem/files.rs:148]: Can we put this function in this existing impl block above?

File: src/volume_mgr.rs [hunk 2]
```diff
@@ -775,6 +775,13 @@ where
 
     /// Close a file with the given full path.
     pub fn close_file(&mut self, file: RawFile) -> Result<(), Error<D::Error>> {
+        let file_idx = self.flush_file(file)?;
+        self.open_files.swap_remove(file_idx);
+        Ok(())
+    }
+
+    /// Flush (update the entry) for a file with the given full path.
```
Comment (thejpster) [inline src/volume_mgr.rs:783]: change "given full path" to "given raw file handle".

Comment (thejpster) [reply src/volume_mgr.rs:783]: It also doesn't make sense to return the internal index here. Maybe make this a private function, and have the public `close_file` and `flush_file` call it?

**Review by thejpster — COMMENTED** [review]
In general, this looks good. Just a few minor notes as I read through.

I wonder also if we can a test which creates a file, writes a few bytes, flushes it, and then checks the directory entry is updated with the new file size.

**Comment by peterkrull** [comment]
@thejpster I have implemented your suggestions

**Comment by thejpster** [comment]
Just a cargo fmt and I think we're good to go.

**Review by thejpster — APPROVED** [review]

**Pull request merged**